backends/bmv2/common/action.cpp

```diff
@@ -74,6 +74,23 @@ void ActionConverter::convertActionBody(const IR::Vector<IR::StatOrDecl>* body,
                         mce2 = new IR::MethodCallExpression(mce->method, mce->typeArguments);
                         mce2->arguments = args;
                         s = new IR::MethodCallStatement(mce);
+                    } else if (em->originalExternType->name.name == "Hash" ||
+                                                em->method->name.name == "get_hash") {
+                        isR = true;
+                        auto dest = new IR::Argument(l);
+                        auto args = new IR::Vector<IR::Argument>();
+                        args->push_back(dest);
+                        if (mce->arguments->size() == 3) {
+                            args->push_back(mce->arguments->at(0));  // base
+                            args->push_back(mce->arguments->at(1));  // data
+                            args->push_back(mce->arguments->at(2));  // max
+                        } else if (mce->arguments->size() == 1) {
+                            args->push_back(mce->arguments->at(0));  // data
+                        } else
+                            BUG("%1%: expected 1 or 3 arguments", mce);
```

Comment:
I would suggest using a modelError here instead of a bug (unless you are sure you have validated the arguments before).